Remove dup start events

UtahDave opened the pull request:
### What does this PR do?
Remove duplicate `minion_start` and `syndic_start` events.

### What issues does this PR fix or reference?
None

### Previous Behavior
When the minion started 2 events with the following event tags were sent up the event bus:

`minion_start`
`salt/minion/<minion id>/start`

The same behavior happened for the salt syndic daemon.

### New Behavior
Only `salt/minion/<minion id>/start` event is sent on minion startup.

The same behavior now happens for the salt syndic daemon.

### Tests written?

No, but I verified that current tests were only testing the new style event tags.
The old style event tags were untested. This PR removes a bit of untested code.

### Commits signed with GPG?

Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

UtahDave committed:
Remove old style `minion_start` event

This removes old style `minion_start` and `syndic_start` events
New style events are still emitted.

UtahDave committed:
Use new style event tags

terminalmage reviewed (commented):
The old events are legacy and had been left for backward compatibility. Have we announced a deprecation plan for removing them?

CC: @cachedout

UtahDave commented:
Are these test failures happening in other PRs on develop?

rallytime commented:
Yeah, they are. I agree with @terminalmage though - I don't think we can just remove these kinds of events without some warning or putting these on a deprecation path.

UtahDave commented:
OK. What kind of warning do we want? A section in the release notes?

terminalmage commented:
@UtahDave yeah, we can put one in the 2018.3.0 release notes and give it the standard 2-release deprecation path, which would mean we'd target the Neon release for final removal.

We could also gate these legacy events behind a config option (defaulting that option to True), which would allow people who operate at large scale and are concerned with the extra overhead incurred by the legacy events to disable them until we can fully deprecate them.

UtahDave committed:
Revert "Remove old style `minion_start` event"

This reverts commit 2062a8c17988cf9923692304701ebde67173c3a7.

UtahDave committed:
Add docs for turning off extra startup msgs

UtahDave committed:
Add docs and defaults for sending old event types

UtahDave committed:
add gate for sending old style startup events

UtahDave committed:
Add minion startup event change to release notes

UtahDave commented:
OK. I've added a gate with a new minion config option instead of just removing the events. I've added a note in the fluorine release notes and opened a github issue to remind us to change the default to False when the Neon release of Salt is made.

https://github.com/saltstack/salt/issues/46336

terminalmage commented on doc/ref/configuration/minion.rst:
```diff
@@ -208,6 +208,28 @@ minion event bus. The value is expressed in bytes.
+.. code-block:: yaml
+
+    enable_legacy_startup_events: True
+
```
:heart: 

Love that you added docs too :smile:

UtahDave replied:
:)  More docs is always good!

terminalmage commented on doc/topics/releases/fluorine.rst:
```diff
@@ -3,3 +3,22 @@
+defaults to ``True``, but will be set to default to ``False`` beginning with
+the Neon release of Salt.
+
+The Salt Syndic currently sends an old style  `sydic_start` event as well. The
```
should be `syndic_start` here.

UtahDave replied:
Ah, great catch.  I'll fix that right now.

terminalmage reviewed (changes requested):
One minor spelling error in the release notes, otherwise this is GTM.

UtahDave committed:
fix typo
